### ISSUE TITLE
Add support to include a `request_reason` with the `X-Goog-Request-Reason` system parameter header

### ISSUE THREAD

**Issue by joaopenteado** [google-github-actions/auth#412]
### TL;DR

It would be very useful for auditing and tracking if we were able to specify a `request_reason` input paramater that is included with every API request in the [`X-Goog-Request-Reason`](https://cloud.google.com/apis/docs/system-parameters).

Users would also be able to dynamically generate this parameter based on the context of the workflow run or outputs from previous steps.

### Detailed design

```yaml
uses: 'google-github-actions/auth@v2'
with:
  project_id: 'my-project'
  workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
  request_reason: 'foo'
```

#### Should a default `request_reason` be provided if none is supplied by the user?
I often like to include the GitHub actions run/job URL and I think it's a reasonable default, but other might differ.
```yaml
uses: 'google-github-actions/auth@v2'
with:
  project_id: 'my-project'
  workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
  request_reason: 'https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs/${{ github.job }}'
```

**Comment by sethvargo** [comment]
Hi @joaopenteado thanks for opening an issue. This is interesting. Which API call(s) do you think would benefit from having this header? I'm also not sure if that header applies to the sts endpoint (which is inherently unauthenticated).
